src/main/java/Floda.java

```diff
@@ -0,0 +1,248 @@
+        Scanner taskScanner = new Scanner(line);
+        taskScanner.next();
+        if (taskScanner.hasNextInt()) {
+            int taskNumber = taskScanner.nextInt() - 1;
+            if (isValidTaskNumber(taskNumber)) {
+                list.get(taskNumber).setDone(false);
+                System.out.println("I have marked this task as not done:\n" + list.get(taskNumber));
+            } else {
+                throw new InvalidInputException("Invalid task number! Please check with 'list'.");
+            }
+        } else {
+            throw new InvalidInputException("Invalid input! Please check with 'list'.");
+        }
+        taskScanner.close();
+    }
+
+    private static void handleDeadlineTask(String line) throws InvalidInputException {
+        Scanner taskScanner = new Scanner(line);
+        taskScanner.next();
+        String remaining = taskScanner.nextLine().trim();
+        int byIndex = remaining.indexOf("/by");
+        if (byIndex == -1 || byIndex == 0) {
+            throw new InvalidInputException("Invalid input format! Use: 'deadline <description> /by <time>'");
+        }
+        String description = remaining.substring(0, byIndex).trim();
+        String by = remaining.substring(byIndex + 3).trim();
+        list.add(new Deadline(description, by, false));
+        System.out.println("Added: " + list.getLast() + "\nNow you have " + list.size() + " items in the list!");
+    }
+
+    private static void handleEventTask(String line) throws InvalidInputException {
+        Scanner taskScanner = new Scanner(line);
+        taskScanner.next();
+        String remaining = taskScanner.nextLine().trim();
+        int fromIndex = remaining.indexOf("/from");
+        if (fromIndex == -1 || fromIndex == 0) {
+            throw new InvalidInputException("Invalid input format! Use: 'event <description> /from <start time> /to <end time>'");
+        }
+        int toIndex = remaining.indexOf("/to");
+        if (toIndex == -1 || toIndex <= fromIndex + 5) {
+            throw new InvalidInputException("Invalid input format! Use: 'event <description> /from <start time> /to <end time>'");
+        }
+        String description = remaining.substring(0, fromIndex).trim();
+        String from = remaining.substring(fromIndex + 5, toIndex).trim();
+        String to = remaining.substring(toIndex + 3).trim();
+        list.add(new Events(description, from, to, false));
+        System.out.println("Added: " + list.getLast() + "\nNow you have " + list.size() + " items in the list!");
+    }
+
+    private static void handleTodoTask(String line) throws InvalidInputException {
+        Scanner taskScanner = new Scanner(line);
+        taskScanner.next();
+        String remaining = taskScanner.nextLine().trim();
+        list.add(new ToDo(remaining, false));
+        System.out.println("Added: " + list.getLast() + "\nNow you have " + list.size() + " items in the list!");
+    }
```

Comment:
Throughout your codebase, there are many magic literals. Avoid them because it is not obvious to the reader what some of them are supposed to mean (readability). Consider `fromIndex + 5`, what '5' is supposed to represent is not immediately clear.